### DISCUSSION TITLE
Check the word after complete all cells

### DISCUSSION THREAD

**Discussion by bhuva** [viresh-ratnakar/exolve#79] [General]
Hello! Thank you for your wonderful script! 
Could you tell me how can I set the script up to check last word as I complete it? Without pressing button 'check this'?
And if there is error with the word where would be a message about it?
Looking forward your answer!

**Comment by viresh-ratnakar** [comment]
I have now added a customization example that you can use as a starting point, in the README file, here:

https://github.com/viresh-ratnakar/exolve/blob/master/README.md#customizations

As for showing a message: well I think the call to checkCurr() will clear incorrect entries, which should be enough. If you really want to add a message, well, it's just javascript, you can change anything. You can change the preamble text, for example.

**Reply by bhuva** [reply]
Thank you! It really check each inputed letter. 
Please tell me - how check whole word (not letter) after complete input?

**Comment by viresh-ratnakar** [comment]
It only checks if every letter has been entered in the current light. If
any entry is not in A-Z, chrckCurr() is not called.

**Reply by bhuva** [reply]
Is there any event when a word completed?

**Comment by viresh-ratnakar** [comment]
No, you will have to create an event, and the logic for that will be
exactly what I've already given you

**Comment by bhuva** [comment]
Thank you for answer! Yes, logic of check the word would be the same, I got it )))) Help me please with - I'm really beginning with JS and I cannot find it in your script exolve-m.js :
1_ user entered last letter of word, so what would be condition that word is complete?
2_ how check the entered word (letters) is correct?
Thank you in advance!

**Reply by viresh-ratnakar** [reply]
Sorry, you'll have to spend some effort learning JS, and I cannot really help with that.

If you create a puzzle starting from, for example, a copy of [exolve-m-simple.html](https://github.com/viresh-ratnakar/exolve/blob/master/exolve-m-simple.html), and you provide solution letters in the puzzle specs, and you add the extra script shown in the [customizations section in README](https://github.com/viresh-ratnakar/exolve#customizations), then you'll get exactly the behaviour you're looking for.

The logic for 1 is this (from the script):
```
      let allEntered = (this.activeCells.length > 0);
      for (let cell of this.activeCells) {
        const row = cell[0];
        const col = cell[1];
        const letter = this.grid[row][col].currLetter;
        if (letter < 'A' || letter > 'Z') {
          allEntered = false;
          break;
        }
      }
      if (allEntered) {
        ...
```
Note that completing an entry need not happen at the last letter of the entry, it can happen anywhere in the entry!

And the logic for 2 is this (also from the script):
```
      if (allEntered) {
        this.checkCurr();
      }
```

**Reply by bhuva** [reply]
Thank you Viresh! I'm learning on the field )))
